self-api/src/main/java/com/selfxdsd/api/storage/Paged.java

```diff
@@ -0,0 +1,66 @@
+package com.selfxdsd.api.storage;
+
+/**
+ * Contains the necessary info for clients to access a {@link Storage} resource
+ * (Projects, Contributors, Tasks etc...) in a paged manner.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.13
+ */
+public interface Paged {
+
+    /**
+     * Current page. Page must be between 1 and totalPages.
+     * @return Page.
+     */
+    Page current();
+
+    /**
+     * Total number of pages.
+     * @return Integer.
+     */
+    int totalPages();
+
+    /**
+     * Data class that encapsulates the number and the size of a page.
+     */
+    final class Page {
+
+        /**
+         * Page number.
+         */
+        private final int number;
+
+        /**
+         * Page size.
+         */
+        private final int size;
+
+        /**
+         * Ctor.
+         * @param number Page number.
+         * @param size Page size.
+         */
+        public Page(final int number, final int size) {
```

Comment:
@criske A page is actually defined by ``index`` and ``size`` -- index where we start the page and size of the page. Maybe your ``number`` is actually the start index?

Reply:
@amihaiemil the page number is the page number starting from 1.
Having a table index based aproach will require the client to know/guess the "table" size imo.
For example if we have 100 records and call `/projects/2/10` then paged implemention will know that it must fetch 10 projects starting from index 10, instead of letting the client to do that logic otherwise.